QtRenderingWidget/ExternFunctions/Creators.cpp

```diff
@@ -2,28 +2,94 @@
 
 #include <exception>
 
-#include <CommonLibs\Logger\logger.h>
-#include "..\..\Rldx\Rldx\Creators\DxSceneCreator.h"
-#include "..\..\Rldx\Rldx\Managers\ResourceManager\DxResourceManager.h"
-#include "..\QtObjects\Views\QtRenderWidgetView.h"
+#include <CommonLibs\Logger\Logger.h>
+
+#include <Rldx\Rldx\Creators\DxSceneCreator.h>
+
+#include <QtRenderingWidget\QtObjects\Views\QtRenderWidgetView.h>
 
 using namespace logging;
 using namespace utils;
 
+static void (*AssetFetchCallBackStored) (QList<QString>* missingFiles, QList<QByteArray>* outBinFiles) = nullptr;
+
+void AssetFetchCallbackWrapper(
+	std::vector<std::wstring>* filesToFetch,
+	std::vector<std::vector<unsigned char>>* outBinFiles
+) {
+
+	// If we don't have callback, use the old debug mode.
+	if (AssetFetchCallBackStored == nullptr) {
+
+		// force list to be same size, maybe redundant
+		outBinFiles->clear();
+		outBinFiles->resize(filesToFetch->size());
+
+		for (const std::wstring& Asset : *filesToFetch) {
+
+			if (IsDiskFile(Asset)) {
+				throw std::exception("Cannot accept disk files");
+			}
+
+			// TODO: the "GetGameAssetFolder" is only for "local callback" debuggin
+			auto filePath = rldx::DxResourceManager::GetGameAssetFolder() + Asset;
+			ByteStream newStream(filePath, false);
+
+
+			if (!newStream.IsValid())
+			{
+				std::vector<unsigned char> dest = std::vector<unsigned char>();
+				outBinFiles->push_back(dest);
+				continue;
+			}
+
+			// -- resize dest, buffer, and read whole file into it
+			std::vector<unsigned char> dest = std::vector<unsigned char>(newStream.GetBufferSize());
+			newStream.Read(dest.data(), dest.size());
+		}
+	}
+	
+	// If we have callback, use it for getting the files.
+	else {
+
+		QList<QString>* filesToFetchQt = new QList<QString>();
+		QList<QByteArray>* outBinFilesQt = new QList<QByteArray>();
+
```

Comment:
Memory leak,should NOT use NEW here

Just declare a stack varable
```
QList<QString> filesToFetchQt;
QList<QByteArray> outBinFilesQt;
```

Reply:
Should be fixed in here: https://github.com/mr-phazer/QtRenderingWidget/pull/3/commits/c3fcd9e335d9dedac08b969e09f1aa1ec01db780